jstests/aggregation/bugs/server7695_isodates.js

```diff
@@ -0,0 +1,269 @@
+// SERVER-7695: Add $isoWeek, $isoWeekYear, and $isoDayOfWeek aggregation expressions.
+
+(function() {
+    "use strict";
+    var coll = db.server7695;
+    var testOpCount = 0;
+
+    coll.drop();
+
+    // Seed collection so that the pipeline will execute.
+    assert.writeOK(coll.insert({}));
+
+    // Helper for testing that op returns expResult.
+    function testOp(op, value, expResult) {
+        testOpCount++;
+        var pipeline = [{$project: {_id: 0, result: {}}}];
+        pipeline[0].$project.result[op] = value;
+        var msg = "Exptected {"+op+": "+value+"} to equal: "+expResult;
+        assert.eq(
+            coll.aggregate(pipeline).toArray()[0].result,
+            [{result: expResult}][0].result,
+            msg);
+    }
+
+    // while development there was a bug which caused an error with $dateToString if the order of
+    // %V and %G changed, so I added this test to prevent regression
+    testOp('$dateToString', { date: new Date("1900-12-31T23:59:59Z"), format: "%V-%G"}, "01-1901")
+    // this is failing, but it shouldn't as it is the same as above, only rotated
```

Comment:
`Value::coerceToTm()` only works on a value, but I just have the year in this Method, because the date wouldn't work. I need the end of the year not just any date in the year. But your debugging output helped a lot and I got it working with `mktime()`